packages/frontend-2/components/viewer/saved-views/panel/views/Group.vue

```diff
@@ -169,6 +175,104 @@ const menuItems = computed((): LayoutMenuItem<MenuItems>[][] => [
   ]
 ])
 
+const onDragOver = (_event: DragEvent) => {
+  _event.preventDefault()
+  if (_event.dataTransfer) {
+    _event.dataTransfer.dropEffect = 'move'
+  }
+}
+
+const onDragEnter = (_event: DragEvent) => {
+  _event.preventDefault()
+  dragCounter.value++
+  isDragOver.value = true
+}
+
+const onDragLeave = (_event: DragEvent) => {
+  dragCounter.value--
+  if (dragCounter.value === 0) {
+    isDragOver.value = false
+  }
+}
+
+const onDrop = async (event: DragEvent) => {
+  event.preventDefault()
+  isDragOver.value = false
+  dragCounter.value = 0
+
+  try {
+    const dataTransfer = event.dataTransfer
+    if (!dataTransfer) return
+
+    const data = JSON.parse(dataTransfer.getData('application/json'))
+    if (!data.viewId || data.sourceGroupId === props.group.id) {
+      return // Same group, no need to move
+    }
+
+    // Move the view to this group using the enhanced drag data
+    const success = await moveViewToGroup(data)
+
+    if (success) {
+      // Auto-open the group if it was closed
+      if (!open.value) {
+        open.value = true
+      }
+
+      triggerNotification({
```

Comment:
we probably want these to always fire whenever a view is moved any way (through dragging or otherwise), so all of this should happen inside of `moveViewToGroup`